### PR TITLE
Add spack

### PR DESCRIPTION
**Pull request by ltalirz**
first try at spack recipe

### CODE REVIEW

**Comment by ltalirz** [comment]
Note: needed to add instructions for fetching `CMakeBuild_FOUND` to gfcc cmakelist (was only present in tamm one)

File: spack/packages/tamm/package.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Tamm(CMakePackage,CudaPackage):
+    """Green's Function Coupled Cluster Library"""
```
Comment (ltalirz) [inline spack/packages/tamm/package.py:10]: what does tamm stand for?

Comment (edoapra) [reply spack/packages/tamm/package.py:10]: Tensor Algebra for Many-body Methods
https://arxiv.org/abs/2201.01257

Comment (ltalirz) [reply spack/packages/tamm/package.py]: Thanks Edoardo!

**Review by ajaypanyala — APPROVED** [review]

**Pull request merged**